### PR TITLE
fix(cb2-9200): fix the Axle Weights not incrementing

### PR DESCRIPTION
**Pull request by JunYanBJSS**
##  Axle weights is not incrementing 

fix the Axle Weights not incrementing
Many thanks for help from Dan and Nathan
[CB2-9200](https://dvsa.atlassian.net/browse/CB2-9200)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/forms/custom-sections/weights/weights.component.ts [hunk 1]
```diff
@@ -103,20 +103,23 @@ export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
...
+    if (this.form && vehicleTechRecord && !this.ladenWeightOverride) {
+      const { currentValue, previousValue } = vehicleTechRecord;
 
       const fieldsChanged = [
         'techRecord_seatsUpperDeck',
         'techRecord_seatsLowerDeck',
         'techRecord_manufactureYear',
-        'techRecord_grossKerbWeight',
-        'techRecord_standingCapacity'
+        'techRecord_grossKerbWeight'
```
Comment (naathanbrown) [inline src/app/forms/custom-sections/weights/weights.component.ts:114]: @Daniel-Searle can you review this? Think there maybe some logic that needs aligning with your changes, such as standing capacity and the manufacuterYear on the IF check

File: src/app/forms/custom-sections/weights/weights.component.ts [hunk 2]
```diff
@@ -103,20 +103,23 @@ export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
...
       ].some(field => currentValue[field] !== previousValue[field]);
 
-      if (fieldsChanged && currentValue.techRecord_manufactureYear) {
-        this.form.patchValue({ techRecord_grossLadenWeight: this.calculateGrossLadenWeight() }, { emitEvent: false });
+      if (fieldsChanged && this.vehicleTechRecord.techRecord_vehicleType === 'psv') {
```
Comment (Daniel-Searle) [inline src/app/forms/custom-sections/weights/weights.component.ts:117]: This if is missing the manufacture year flag, we only want to auto calculate the year of manufacture when there is a manufacture year present

File: src/app/forms/custom-sections/weights/weights.component.ts [hunk 3]
```diff
@@ -103,20 +103,23 @@ export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   private handleVehicleTechRecordChange(changes: SimpleChanges): void {
-    if (changes['vehicleTechRecord'] && !this.ladenWeightOverride && this.form) {
-      const { currentValue, previousValue } = changes['vehicleTechRecord'];
+    const { vehicleTechRecord } = changes;
+    if (this.form && vehicleTechRecord && !this.ladenWeightOverride) {
+      const { currentValue, previousValue } = vehicleTechRecord;
 
       const fieldsChanged = [
```
Comment (Daniel-Searle) [inline src/app/forms/custom-sections/weights/weights.component.ts:110]: You've missing the standing capacity from the fields change. With it removed, it isn't going to re-calculate when it changes

File: src/app/forms/custom-sections/weights/weights.component.ts [hunk 4]
```diff
@@ -103,20 +103,23 @@ export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   private handleVehicleTechRecordChange(changes: SimpleChanges): void {
-    if (changes['vehicleTechRecord'] && !this.ladenWeightOverride && this.form) {
-      const { currentValue, previousValue } = changes['vehicleTechRecord'];
+    const { vehicleTechRecord } = changes;
```
Comment (Daniel-Searle) [inline src/app/forms/custom-sections/weights/weights.component.ts:106]: Why has this been removed from the inline if?

File: src/app/forms/custom-sections/weights/weights.component.ts [hunk 5]
```diff
@@ -103,20 +103,23 @@ export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
...
         'techRecord_manufactureYear',
-        'techRecord_grossKerbWeight',
-        'techRecord_standingCapacity'
+        'techRecord_grossKerbWeight'
       ].some(field => currentValue[field] !== previousValue[field]);
 
-      if (fieldsChanged && currentValue.techRecord_manufactureYear) {
-        this.form.patchValue({ techRecord_grossLadenWeight: this.calculateGrossLadenWeight() }, { emitEvent: false });
+      if (fieldsChanged && this.vehicleTechRecord.techRecord_vehicleType === 'psv') {
+        const newGrossLadenWeight = this.calculateGrossLadenWeight();
```
Comment (Daniel-Searle) [inline src/app/forms/custom-sections/weights/weights.component.ts:118]: You could probably inline this variable

File: src/app/forms/custom-sections/weights/weights.component.ts [hunk 6]
```diff
@@ -103,20 +103,23 @@ export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
...
+        this.vehicleTechRecord.techRecord_grossLadenWeight = newGrossLadenWeight;
```
Comment (Daniel-Searle) [inline src/app/forms/custom-sections/weights/weights.component.ts:119]:
```suggestion
        this.vehicleTechRecord.techRecord_grossLadenWeight = this.calculateGrossLadenWeight();

```

File: src/app/forms/custom-sections/weights/weights.component.ts (hunk 4 above)
Comment (JunYanBJSS) [reply src/app/forms/custom-sections/weights/weights.component.ts:106]: I added it back.

File: src/app/forms/custom-sections/weights/weights.component.ts [hunk 7]
```diff
@@ -114,9 +115,13 @@ export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
         'techRecord_standingCapacity'
       ].some(field => currentValue[field] !== previousValue[field]);
 
-      if (fieldsChanged && currentValue.techRecord_manufactureYear) {
-        this.form.patchValue({ techRecord_grossLadenWeight: this.calculateGrossLadenWeight() }, { emitEvent: false });
+      if (fieldsChanged && currentValue.techRecord_manufactureYear) {
```
Comment (naathanbrown) [inline src/app/forms/custom-sections/weights/weights.component.ts:118]: This can be made into one if

**Review by naathanbrown — APPROVED** [review]

**Review by Daniel-Searle — APPROVED** [review]

**Pull request merged**